src/contexts/AzureMapLayerContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { Component, createContext, useContext, useState, useEffect } from 'react'
+import {
+  IAzureMapLayerProps,
+  IAzureLayerStatefulProviderProps,
+  IAzureMapsContextProps
+} from '../types'
+import atlas from 'azure-maps-control'
+import { AzureMapsContext } from './AzureMapContext'
+import { IAzureMapDataSourceProps } from '../types'
+import { AzureMapDataSourceContext } from './AzureMapDataSourceContext'
+
+const AzureMapLayerContext = createContext<IAzureMapLayerProps>({
+  layerRef: null
+})
+const { Provider, Consumer: AzureMapLayerConsumer } = AzureMapLayerContext
+
+const AzureMapLayerStatefulProvider = ({ id, options }: IAzureLayerStatefulProviderProps) => {
+  const { mapRef } = useContext<IAzureMapsContextProps>(AzureMapsContext)
+  const { dataSourceRef } = useContext<IAzureMapDataSourceProps>(AzureMapDataSourceContext)
+  const [layerRef, setLayerRef] = useState<atlas.layer.SymbolLayer | null>(null)
+
+  useEffect(() => {
+    if (mapRef && layerRef && dataSourceRef) {
+      setLayerRef(new atlas.layer.SymbolLayer(dataSourceRef, id, options))
+      mapRef.layers.add(layerRef)
+      return () => {
+        mapRef.layers.remove(layerRef)
+      }
+    }
+  }, [])
+
+  return (
+    <Provider
+      value={{
+        layerRef
+      }}
+    ></Provider>
```

Comment:
Is this Layer provider will be without any children?

Reply:
As I see in the previous implementation layer was used only to add the DataSource, we can adjust the implementation if we needed to